### PR TITLE
Queue discards elements with same ID and same type which are _older_.

### PR DESCRIPTION
**Pull request by kovacshuni**

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -13,14 +16,27 @@ func newUnique(capacity int) uniqueue {
 }
 
 func (cb *uniqueue) enqueue(n *notificationUPP) {
-	wasRemoved := true
-	for wasRemoved {
-		wasRemoved = false
-		for i, e := range cb.buffer {
-			if e.ID == n.ID && e.Type == n.Type {
+	if n.LastModified == "" {
```
Comment (tosan88) [inline queue.go]: is it worth to add a test for this case too? :-?

**Comment by tosan88** [comment]
:+1:

File: queue.go [hunk 2]
```diff
@@ -13,14 +16,27 @@ func newUnique(capacity int) uniqueue {
...
+	}
+	for i, e := range cb.buffer {
+		if e.ID == n.ID && e.Type == n.Type {
+			eDate, err := time.Parse(time.RFC3339Nano, e.LastModified)
+			if err != nil {
+				warnLogger.Printf("Notification in cache has malformed date: %v", n)
+				return
+			}
+			if eDate.Before(newestRawDate) {
 				cb.buffer = append(cb.buffer[:i], cb.buffer[i+1:]...)
```
Comment (Nesze) [inline queue.go]: It does not belong to this PR, but is there a potential race condition?

**Comment by Nesze** [comment]
👍
